test/monitoring/aws-ec2/EC2Monitoring.test.ts

```diff
@@ -2,70 +2,90 @@ import { Stack } from "aws-cdk-lib";
 import { Template } from "aws-cdk-lib/assertions";
 import { AutoScalingGroup } from "aws-cdk-lib/aws-autoscaling";
 
-import { EC2Monitoring } from "../../../lib";
+import { EC2Monitoring, InstanceFamilyTypes } from "../../../lib";
 import { addMonitoringDashboardsToStack } from "../../utils/SnapshotUtil";
 import { TestMonitoringScope } from "../TestMonitoringScope";
 
-test("snapshot test: all instances, no alarms", () => {
-  const stack = new Stack();
-
-  const scope = new TestMonitoringScope(stack, "Scope");
-
-  const monitoring = new EC2Monitoring(scope, {
-    alarmFriendlyName: "EC2",
-  });
-
-  addMonitoringDashboardsToStack(stack, monitoring);
-  expect(Template.fromStack(stack)).toMatchSnapshot();
-});
-
-test("snapshot test: ASG, no alarms", () => {
-  const stack = new Stack();
-
-  const scope = new TestMonitoringScope(stack, "Scope");
-
-  const monitoring = new EC2Monitoring(scope, {
-    alarmFriendlyName: "EC2",
-    autoScalingGroup: AutoScalingGroup.fromAutoScalingGroupName(
-      stack,
-      "DummyASG",
-      "DummyASG"
-    ),
-  });
-
-  addMonitoringDashboardsToStack(stack, monitoring);
-  expect(Template.fromStack(stack)).toMatchSnapshot();
-});
-
-test("snapshot test: instance filter, no alarms", () => {
-  const stack = new Stack();
-
-  const scope = new TestMonitoringScope(stack, "Scope");
-
-  const monitoring = new EC2Monitoring(scope, {
-    alarmFriendlyName: "EC2",
-    instanceIds: ["instance1", "instance2"],
-  });
-
-  addMonitoringDashboardsToStack(stack, monitoring);
-  expect(Template.fromStack(stack)).toMatchSnapshot();
-});
-
-test("snapshot test: instance filter + ASG, no alarms", () => {
-  const stack = new Stack();
-
-  const scope = new TestMonitoringScope(stack, "Scope");
-
-  const monitoring = new EC2Monitoring(scope, {
-    alarmFriendlyName: "EC2",
-    instanceIds: ["instance1", "instance2"],
-    autoScalingGroup: AutoScalingGroup.fromAutoScalingGroupName(
-      stack,
-      "DummyASG",
-      "DummyASG"
-    ),
-  });
-
-  addMonitoringDashboardsToStack(stack, monitoring);
-  expect(Template.fromStack(stack)).toMatchSnapshot();
-});
+function enumKeys<O extends object, K extends keyof O = keyof O>(obj: O): K[] {
+  return Object.keys(obj).filter((k) => Number.isNaN(+k)) as K[];
+}
+
+test.each(enumKeys(InstanceFamilyTypes))(
```

Comment:
You could just iterate over the enum values once and do the tests rather than iterating repeatedly for each test.

Reply:
Do you mean not parametrizing the tests and have a single test instead?

Reply:
I mean doing something like

```ts
enumKeys(InstanceFamilyTypes).forEach(familyType => {
  test(...) {}
  
  test(...) {}
  
  test(...) {}
});
```